File: android/src/main/kotlin/com/tokoin/pos_printer/model/Enum.kt
package com.tokoin.pos_printer.model

inline fun <reified T : Enum<*>> enumValueOrNull(name: String): T? =
    T::class.java.enumConstants?.firstOrNull { it.name.equals(name, ignoreCase = true) }

enum class POSCommandType {
    Text,
    KeyValue,
    Linebreak,
    Divider,
    Image
}

enum class POSAlign {
    Left,
    Right,
    Center
}

enum class POSFont {
    Small,
    Medium,
    Wide,
    Tall,
    Big
}

enum class POSFontWeight {
    Normal,
    Bold
}

enum class POSUnderline {
    None,
    Single,
    DoubleStrike
}

enum class POSWidth {
    Full,
    Half
}

File: lib/model/enum.dart
import 'package:json_annotation/json_annotation.dart';

enum POSCommandType {
  @JsonValue('Text')
  text,
  @JsonValue('KeyValue')
  keyValue,
  @JsonValue('Linebreak')
  linebreak,
  @JsonValue('Divider')
  divider,
  @JsonValue('Image')
  image
}
enum POSAlign {
  @JsonValue('Left')
  left,
  @JsonValue('Right')
  right,
  @JsonValue('Center')
  center
}

enum POSFont {
  @JsonValue('Small')
  small,
  @JsonValue('Medium')
  medium,
  @JsonValue('Wide')
  wide,
  @JsonValue('Tall')
  tall,
  @JsonValue('Big')
  big
}

enum POSFontWeight {
  @JsonValue('Normal')
  normal,
  @JsonValue('Bold')
  bold
}

enum POSUnderline {
  @JsonValue('None')
  none,
  @JsonValue('Single')
  single,

  @JsonValue('DoubleStrike')
  doubleStrike
}

enum POSWidth {
  @JsonValue('Full')
  full,
  @JsonValue('Half')
  half
}
